---
title: "R Notebook"
output:
  html_document:
    df_print: paged
---

```{r}

library(edgeR)
library(Mus.musculus)
library(GlimmaV2)

files <- c("GSM1545535_10_6_5_11.txt", "GSM1545536_9_6_5_11.txt", 
   "GSM1545538_purep53.txt", "GSM1545539_JMS8-2.txt", 
   "GSM1545540_JMS8-3.txt", "GSM1545541_JMS8-4.txt", 
   "GSM1545542_JMS8-5.txt", "GSM1545544_JMS9-P7c.txt", 
   "GSM1545545_JMS9-P8c.txt")

x <- readDGE(files, columns=c(1,3))
class(x)
```

```{r}
dim(x)
```

```{r}
samplenames <- substring(colnames(x), 12, nchar(colnames(x)))
colnames(x) <- samplenames
group <- as.factor(c("LP", "ML", "Basal", "Basal", "ML", "LP", 
                     "Basal", "ML", "LP"))
x$samples$group <- group
lane <- as.factor(rep(c("L004","L006","L008"), c(3,4,2)))
x$samples$lane <- lane
x$samples
```

```{r}
geneid <- rownames(x)
genes <- select(Mus.musculus, keys=geneid, columns=c("SYMBOL", "TXCHROM"), 
                keytype="ENTREZID")
genes <- genes[!duplicated(genes$ENTREZID),]
x$genes <- genes
x
```


```{r}

# transformations from raw scale

cpm <- cpm(x)
lcpm <- cpm(x, log=TRUE)
summary(lcpm)

```

```{r}

# remove lowly expressed genes
keep.exprs <- filterByExpr(x, group=group)
x <- x[keep.exprs,, keep.lib.sizes=FALSE]
dim(x)
```

```{r}
# normalising gene expression distributions
x <- calcNormFactors(x, method = "TMM")
x$samples$norm.factors

```



```{r}
glimmaMDS(lcpm, labels=paste(group, lane, sep="_"), 
          groups=x$samples)
```

```{r}
library(RColorBrewer)

lcpm <- cpm(x, log=TRUE)
par(mfrow=c(1,2))
col.group <- group
levels(col.group) <-  brewer.pal(nlevels(col.group), "Set1")
col.group <- as.character(col.group)
col.lane <- lane
levels(col.lane) <-  brewer.pal(nlevels(col.lane), "Set2")
col.lane <- as.character(col.lane)
plotMDS(lcpm, labels=group, col=col.group)
title(main="A. Sample groups")
plotMDS(lcpm, labels=lane, col=col.lane, dim=c(3,4))
title(main="B. Sequencing lanes")

```


```{r}

design <- model.matrix(~0+group+lane)
colnames(design) <- gsub("group", "", colnames(design))
design

```

```{r}
contr.matrix <- makeContrasts(
   BasalvsLP = Basal-LP, 
   BasalvsML = Basal - ML, 
   LPvsML = LP - ML, 
   levels = colnames(design))
contr.matrix
```

```{r}

par(mfrow=c(1,2))
v <- voom(x, design, plot=TRUE)
vfit <- lmFit(v, design)
vfit <- contrasts.fit(vfit, contrasts=contr.matrix)
efit <- eBayes(vfit)
plotSA(efit, main="Final model: Mean-variance trend")

```



```{r}

tfit <- treat(vfit, lfc=1)
dt <- decideTests(tfit)
summary(dt)

```

```{r}
class(tfit)
glimmaMA(tfit, coef=2, status=dt, main=colnames(tfit)[2])

```

