---
title: "TestXYPlot"
output: html_document
author: "Hasaru Kariyawasam"
---


```{r}

### TEST: glimmav1 dataset ###

library(Glimma)
library(limma)
library(GlimmaV2)

data(lymphomaRNAseq)
rnaseq <- lymphomaRNAseq

# add lane
groups <- data.frame(genotype=rnaseq$samples$group,
                     lane= as.character(c(rep(4,5),3,3)),
                     miscCont=c(rep(4000,5),300,250),
                     miscDisc=c("blue","red",rep("green",5)))

# add libsize
groups <- rnaseq$samples$group

# fit
design <- model.matrix(~0+groups)
contrasts <- cbind(Smchd1null.vs.WT=c(-1,1))

# convert raw counts to logCPM values by automatically extracting libsizes and normalisation factors from x
vm <- voomWithQualityWeights(rnaseq, design=design)
fit <- lmFit(vm, design=design)
fit <- contrasts.fit(fit, contrasts)
fit <- eBayes(fit)
dtFit <- decideTests(fit)
```

```{r}
# baseline
counts = rnaseq$counts
groups = rnaseq$samples$group
glimmaXY(x=fit$coef, y=fit$lod, status=dtFit, status.colours=c("cyan", "grey", "hotpink"))
```



```{r}
# xlab, ylab added
glimmaXY(x=fit$coef, y=fit$lod, xlab="logFC", ylab="logodds", status=dtFit)
```


```{r}
# xlab, ylab and anno
glimmaXY(x=fit$coef, y=fit$lod, xlab="logFC", ylab="logodds", status=dtFit, anno=fit$genes)
```

